src/test/regress/sql/multi_subquery_behavioral_analytics.sql

```diff
@@ -1600,5 +1618,47 @@ FROM (
 SET client_min_messages TO DEFAULT;
 DROP FUNCTION volatile_func_test();
 
-SET citus.subquery_pushdown to OFF;
+CREATE FUNCTION test_join_function_2(integer, integer) RETURNS bool
+    AS 'select $1 > $2;'
+    LANGUAGE SQL
+    IMMUTABLE
+    RETURNS NULL ON NULL INPUT;
+
+-- we don't support joins via functions 
+SELECT user_id, array_length(events_table, 1)
+FROM (
+  SELECT user_id, array_agg(event ORDER BY time) AS events_table
+  FROM (
+    SELECT u.user_id, e.event_type::text AS event, e.time
+    FROM users_table AS u,
+         events_table AS e
+    WHERE test_join_function_2(u.user_id, e.user_id)
+  ) t
+  GROUP BY user_id
+) q
+ORDER BY 2 DESC, 1;
+
+-- note that the following query has joins on the partition keys
+-- however we fail to push down it due to the function call on the
+-- where clause. We probably need to relax that check
```

Comment:
I think keeping the tests is useful given that in the above one there is actually no join on the partition keys and join is done via the function. In the below one there is actually join on the partition key but we still error out (which we'll probably consider removing in the future).

> you can move these comments above.

Sure, I can do that.